### ISSUE TITLE
Error while loading Registry Key 

### ISSUE THREAD

**Issue by AdmiringWorm** [MASGAU/MASGAU#77]
I recently started getting a RegistryKeyLoadError on the startup of MASGAU (it then closes after the error).
This seems to be a problem with the "Steam App" registry keys, and the new registry security twists of Windows 8 (not 100% sure regarding Windows 8 though).
It only happens when the game with that reg key is installed, and if the entry is deleted MASGAU starts up as normal.
Running MASGAU Version 1.0.4.
Stacktrace below:

```
RegistryKeyLoadError

   at MASGAU.Location.SystemLocationHandler.getPaths(LocationRegistry get_me) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Windows\Location\SystemLocationHandler.cs:line 443
   at MASGAU.Location.ALocationsHandler.getPaths(ALocation get_me) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\Location\ALocationsHandler.cs:line 113
   at MASGAU.GameEntry.Detect() in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\Game\GameEntry.cs:line 287
   at MASGAU.Games.detectGames(List`1 these_games) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\Game\Games.cs:line 297
   at MASGAU.Main.MainProgramHandler.detectGames() in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\AMainProgramHandler.cs:line 42
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
startIndex cannot be larger than length of string.
Parameter name: startIndex

   at System.String.InternalSubStringWithChecks(Int32 startIndex, Int32 length, Boolean fAlwaysCopy)
   at System.String.Substring(Int32 startIndex)
   at MASGAU.Location.ALocationHandler.interpretPath(String interpret_me) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\Location\ALocationHandler.cs:line 184
   at MASGAU.Location.ALocationsHandler.interpretPath(String[] interpret_me) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Common\Location\ALocationsHandler.cs:line 178
   at MASGAU.Location.SystemLocationHandler.getPaths(LocationRegistry get_me) in C:\Users\Matthew\Desktop\MASGAU\MASGAU.Windows\Location\SystemLocationHandler.cs:line 435
```

And as usual the WTF error if I'm trying to send it:

```
WTF?

   at Email.WPF.EmailButton.checkEmails() in C:\Users\Matthew\Desktop\MASGAU\Libs\Email.WPF\EmailButton.xaml.cs:line 82
   at Email.WPF.EmailButton.button_Click(Object sender, RoutedEventArgs e) in C:\Users\Matthew\Desktop\MASGAU\Libs\Email.WPF\EmailButton.xaml.cs:line 37
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)

```

**Comment by sanmadjack** [comment]
I started getting this too. I think Steam changed something.

**Comment by AdmiringWorm** [comment]
hmm, perhaps. I can't find the keys in the registry so steam could have locked the section down. I thought it was a windows 8 problem since (as I said above) they changed something (not sure what) regarding the security settings of the registry.
May I offer a suggestion, move those registry keys into a separate app path (for instance steamreg) and only look it up if nothing else has been found. (Just thinking, so you wouldn't have to use a try catch instead)

**Comment by sanmadjack** [comment]
Are you still getting this error?

**Comment by sanmadjack** [comment]
I've got a possible fix for this error, but I need to confirm it. Do you still get the error?

**Comment by AdmiringWorm** [comment]
Yes, I am. I tried the the latest source for 1.0.5 and the problem is still there with the exact error.

**Comment by AdmiringWorm** [comment]
It seems it failes in ALocationhandler on line 184.

**Comment by AdmiringWorm** [comment]
I now see why it fails. the interpret_me path is 'F:\SteamLibrary\SteamApps\common\4x4 Hummer',
while the folder path is 'D:\Program Files (x86)\Steam\steamapps\common'.
It failes because the folder length is longer than interpret_me so the substring is out of range.
Apparently it hasn't had anything to do with the registry key after all.

**Comment by sanmadjack** [comment]
I haven't checked in my fix yet. This confirms the issue though. I'll check in the code now.

**Comment by sanmadjack** [comment]
Code checked in. Please update and try again.

**Comment by AdmiringWorm** [comment]
Sorry for the late reply, I can confirm now that this is fixed in 1.0.5

**Comment by sanmadjack** [comment]
Thanks much!
